scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/ClientHandshaker.java

```diff
@@ -154,22 +156,25 @@ public ClientHandshaker(DTLSSession session, RecordLayer recordLayer, SessionLis
 		this.preferredCipherSuites = config.getSupportedCipherSuites();
 		this.maxFragmentLengthCode = config.getMaxFragmentLengthCode();
 		this.supportedServerCertificateTypes = new ArrayList<>();
-		this.supportedServerCertificateTypes.add(CertificateType.RAW_PUBLIC_KEY);
-		if (rootCertificates != null && rootCertificates.length > 0) {
-			int index = config.isSendRawKey() ? 1 : 0;
-			this.supportedServerCertificateTypes.add(index, CertificateType.X_509);
-		}
-
 		this.supportedClientCertificateTypes = new ArrayList<>();
-		if (privateKey != null && publicKey != null) {
-			if (certificateChain == null || certificateChain.length == 0) {
-				this.supportedClientCertificateTypes.add(CertificateType.RAW_PUBLIC_KEY);
-			} else if (config.isSendRawKey()) {
-				this.supportedClientCertificateTypes.add(CertificateType.RAW_PUBLIC_KEY);
-				this.supportedClientCertificateTypes.add(CertificateType.X_509);
-			} else {
-				this.supportedClientCertificateTypes.add(CertificateType.X_509);
-				this.supportedClientCertificateTypes.add(CertificateType.RAW_PUBLIC_KEY);
+
+		if (config.useEcdheEcdsa()) {
```

Comment:
So, the idea here is that we only need to indicate preference for any particular types of certificates if we are supporting a cipher suite that requires the exchange of one or more certificates, right?
If so, then I would rather rename the `config.useEcdheEcdsa` method to something that better reflects the purpose. There are more cipher suites that require the exchange of certificates, in particular the ones that we do not support yet which are based on RSA but which have been asked about in the past already. What about something like `isCipherRequiringCertificateExchangeSupported`?

Reply:
BTW the CLIENT_HELLO also includes extensions that are only relevant for ECC based ciphers. I think we should make inclusion of those also dependent on the support for ECC based ciphers ...

Reply:
Uups, I just forgot to pick that commit. I will do that in the evening.